tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -390,6 +367,11 @@ private void AllExternalTools_CollectionChanged(object? sender, NotifyCollection
 
     public void SetRegionsForTitleBar()

```

Comment:
Hmm... if this is handling LayoutUpdated, and you're changing the sizes/positions of things, are we sure there isn't a case where we end up recursing forever?

Also, is there any way to throttle or scope this, as it seems a very active event...?

Reply:
We're not updating sizes/positions, just telling WinUI3 what parts of the titlebar have controls that get input.

Unfortunately I'm not aware of a good way to throttle or scope it. :(

Reply:
I added a little more throttling.... should be much better. Just has a null check and a float comparison for the majority of cases.